src/app/products/products.service.ts

```diff
@@ -112,6 +112,24 @@ export class ProductsService {
     return this.http.get(`/charges/${chargeId}`);
   }
 
+  getSelectedCharge(selectedCharge: string, template: boolean = false): Observable<any> {
```

Comment:
What is the need to define a new function? You can get the charge data through getCharge() function too.